### ISSUE TITLE
RKS+UKS doesn't run in the same input, but UKS+RKS does

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#2174]
Running this input results in a PSIO error in version 1.3.2.
```
molecule h2o {
0 1
O     0.000000     0.000000     0.000000
H     0.000000     0.000000     0.956914
H     0.926363     0.000000    -0.239868
}

molecule ch3 {
0 2
C     0.000000    0.000000     0.000000
H     0.000000    1.078000     0.000000
H     0.933000   -0.539000     0.000000
H    -0.933000   -0.539000     0.000000
}

set basis 6-31G*

activate(h2o)
set reference rhf
energy('b3lyp')

activate(ch3)
set reference uhf
energy('b3lyp')
```

However, if one switches the runs the other way around
```
molecule h2o {
0 1
O     0.000000     0.000000     0.000000
H     0.000000     0.000000     0.956914
H     0.926363     0.000000    -0.239868
}

molecule ch3 {
0 2
C     0.000000    0.000000     0.000000
H     0.000000    1.078000     0.000000
H     0.933000   -0.539000     0.000000
H    -0.933000   -0.539000     0.000000
}

set basis 6-31G*

activate(ch3)
set reference uhf
energy('b3lyp')

activate(h2o)
set reference rhf
energy('b3lyp')
```
the calculation succeeds.

**Comment by hokru** [comment]
Does adding a `clean()` after the first call help?

**Comment by JonathonMisiewicz** [comment]
@susilehtola I cannot reproduce the failure on current master. Please try the _first_ input again and report back with whether the calculation succeeds. If it does, I'm going to close the issue.

**Comment by susilehtola** [comment]
Confirmed; while the first input crashes in 1.3.2 the run is successful in master so this is fixed by Psi4 1.4

**Comment by JonathonMisiewicz** [comment]
Excellent! Closing.
